packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -286,9 +297,32 @@ contract SystemConfig is OwnableUpgradeable, ISemver {
     }
 
     /// @notice Updates the unsafe block signer address. Can only be called by the owner.
+    /// @param _batcherHash New batch hash.
     /// @param _unsafeBlockSigner New unsafe block signer address.
-    function setUnsafeBlockSigner(address _unsafeBlockSigner) external onlyOwner {
+    function setSequencer(bytes32 _batcherHash, address _unsafeBlockSigner) external onlyOwner {
         _setUnsafeBlockSigner(_unsafeBlockSigner);
+        _setBatcherHash(_batcherHash);
+
+        Types.SequencerKeys memory _sequencer =
+            Types.SequencerKeys({ unsafeBlockSigner: _unsafeBlockSigner, batcherHash: _batcherHash });
```

Comment:
@tynes Something you may notice is that `setSequencer()` could accept as struct arg, which I think would be cleaner, ie.

```
setSequencer(Types.SequencerKeys memory _sequencer)
```

However when I put it into the interface, the [same type](https://github.com/ethereum-optimism/optimism/pull/7310/files#diff-e463331be751e9c6fa0f44ee6b7534067c0db4728537611f6575b328fe533b26R32-R35) which is currently in `superchainconfig.go` is also added to `systemconfig.go` and I get this error: 

```
./op-bindings/bindings/systemconfig.go:52:6: TypesSequencerKeys redeclared in this block
../op-bindings/bindings/superchainconfig.go:32:6: other declaration of TypesSequencerKeys
```

I wonder if you've encountered that before?

Reply:
I did have a problem similar to this when doing the gas config in the system config/resource metering contracts. I can't remember exactly how I handled it to be honest. It is trying to generate the type twice in the same package, this is a problem with how we namespace our bindings

Reply:
Ah i understand, its trying to generate the type that is the input to the function and its not smart enough to know that the type is already generated in a different file

Reply:
The exclude functionality on abigen could work, there is also a combined json flag on abigen that we don't use and i wonder if that would help, altho idk how to get the combined json out of forge plus it would prob include the test files and stuff

Reply:
My sense for now its something that we just leave as is and then we solve it sometime in the future

Reply:
Would it be too brittle to find and delete the duplicate type def from one of the bindings at the end of generation?

Reply:
Going to resolve this. In a future clean up I might handle this with --exc